PythonCodeConventions.md

```diff
@@ -1,24 +1,33 @@
 Python Code Conventions
 =======
 
-PEP8 is the code style guide for Python code used by the main python distribution.
-This style guide can be found at:
-https://www.python.org/dev/peps/pep-0008/
+General
+---
+Ultimaker adheres to the PEP8 coding guidelines as described here: https://www.python.org/dev/peps/pep-0008/,
+with the following adjustments, exceptions and specific choices:
 
-Our coding conventions follow the PEP style guide, with the exceptions stated in the generic code_conventions.md provided in this repository. This also means that for rules regarding class naming, documentation, etc., the rules described in the general code convention document MUST be used.
-
-For convenience, these are re-listed here. In case of conflict, the generic code_conventions are considered to be leading.
-* Always use spaces. While PEP8 allows tabs for backwards compatibility, we do not.
-* Strings are double quotes. While Python allows single and double quoted strings, and PEP8 only says "pick a rule and stick to it". We decided to do double quotes to match C++.
-* "Maximum line length" - PEP8 specifies a maximum line length of 79 characters. At Ultimaker we have no official line length limit, although we strive to keep lines below 120 characters.
-* Naming convention:
-  * Function names in PEP8 are in lower_case_with_underscores. At Ultimaker we decided to match our C++ coding convention and use mixedCase (CapitalizedWords by initial lowercase character).
-  * Parameters, variables and members follow the lower_case_with_underscores convention. PEP8 has no recommendation on this.
+* Always use (4) spaces instead of tabs.
+* Strings use double quotes to match C++ style.
+* Maximum line length is set to 120 characters. See the 'line breaking' section below for specific suggested PEP8 style choices.
+* String formatting, multiple elements:
+``` python
+"{foo}_{bar}".format(foo="foo", bar="bar")
+```
+* String formatting, single element:
+``` python
+"{}_bar".format("foo")
+```
+* Naming convention(s):
+ * PEP8 specifies functions to be in lower_case_with_underscores. At Ultimaker we decided to match our C++ coding convention and use mixedCase (CapitalizedWords by initial lowercase character)
+ * Parameters, variables and members follow the lower_case_with_underscores convention. PEP8 has no recommendation on this.
```

Comment:
Why does it matter that pep does/does not have something to say about this?

Reply:
Also, i think we should use the same style as the code convention document (eg; MUST, SHOULD, SHOULD NOT, etc.) That way it's clear what we mean by this.

Reply:
1) to make a distinction between a deviation from PEP8 or a specific choice by Ultimaker without any explicit PEP8 'support'. This text was taken as-is from the previous version.
2) Good idea, will rewrite accordingly.

Reply:
Rewritten.